### PR TITLE
Add methods and tests for listAllGames through endpoint

### PR DESCRIPTION
**Pull request by samBAM089**

### CODE REVIEW

File: backend/src/test/java/de/sambam/gamenighttracker/controller/GameControllerTest.java [hunk 1]
```diff
@@ -53,6 +56,26 @@ public void addNewGameTest() {
...
+                "http://localhost:" + port + "/games", Game[].class);
+        Game[] allTheGamesinDb = getResponse.getBody();
+        Game expectedGame1 = Game.builder().name("Cluedo").build();
+        Game expectedGame2 = Game.builder().name("Mikado").build();
 
+        //THEN
+        assertThat(getResponse.getStatusCode(), is(HttpStatus.OK));
+        assertEquals(2, allTheGamesinDb.length);
+        assertEquals(expectedGame1, allTheGamesinDb[0]);
+        assertEquals(expectedGame2, allTheGamesinDb[1]);
```
Comment (jamarob) [inline backend/src/test/java/de/sambam/gamenighttracker/controller/GameControllerTest.java:79]: ok, but we can write this more elegantly using hamcrest matchers, maybe have a look at those.

File: backend/src/test/java/de/sambam/gamenighttracker/service/GameServiceTest.java [hunk 2]
```diff
@@ -23,4 +28,27 @@ public void addNewGameTester() {
...
     }
-}
\ No newline at end of file
+
+    @Test
+    @DisplayName("listAllGames should return all the games from db")
+    public void listAllGamesTest() {
+        //GIVEN
+        GameDb gameDb = new GameDb();
+        gameDb.add(Game.builder().name("CloudAge").releaseYear("2020").build());
```
Comment (jamarob) [inline backend/src/test/java/de/sambam/gamenighttracker/service/GameServiceTest.java:37]: Better cut the db away and mock it here so you can test the service in isolation

**Review by jamarob — CHANGES_REQUESTED** [review]
Please mock the database in the service test otherwise, lgtm 👍

File: backend/src/test/java/de/sambam/gamenighttracker/controller/GameControllerTest.java (hunk 1 above)
Comment (samBAM089) [reply backend/src/test/java/de/sambam/gamenighttracker/controller/GameControllerTest.java:79]: thanks. i have tried those but it did not work for me in this case... i will have another look after some more progress of the project.

**Review by jamarob — APPROVED** [review]
Excellent! 🏅

**Pull request merged**